### PR TITLE
Fix Extensions on ElementDefs

### PR DESCRIPTION
**Pull request by ngfreiter**
This fixes https://github.com/FHIR/sushi/issues/759.

The issue here was not actually related to child profiles, as is pointed out in the most recent comment. The issue was just that when we call the `cleanResource` function to clean a StructureDefinition of all of the extra properties that SUSHI puts on it, we ignored the `elements` array, because we didn't think those extra properties would end up there. But, in this example:
```
Profile: MyProfile
Parent: Patient
// This does not serialize correctly
* name ^extension[MyBooleanExtension].valueBoolean = true


Extension: MyBooleanExtension
* value[x] only boolean
```
Those elements do end up in the `elements` array, and are not correctly cleaned. This PR fixes that by now cleaning the `elements` array. We do still have to avoid the `structDef` element that exists on an ElementDefinition, because otherwise we will infinitely recurse.

You can test this using the example above, or by running the test on https://github.com/FHIR/sushi/commit/52609f626d8a8c8dbad9ec927cd79e10c94d9da5, where it should fail.

### CODE REVIEW

**Review by julianxcarter — DISMISSED** [review]
LGTM!

File: src/export/StructureDefinitionExporter.ts [hunk 1]
```diff
@@ -567,7 +567,7 @@ export class StructureDefinitionExporter implements Fishable {
     // The elements list does not need to be cleaned up.
```
Comment (jafeltra) [inline src/export/StructureDefinitionExporter.ts:567]: I might owe you something for this comment, but I don't think we need this line anymore.

**Review by jafeltra — COMMENTED** [review]
I ran the regression on the selected repos and didn't see any changes! I did leave one _very_ small comment 😰

File: src/export/StructureDefinitionExporter.ts (hunk 1 above)
Comment (ngfreiter) [reply src/export/StructureDefinitionExporter.ts]: For 4 days I have toiled to fix this issue. But at long last, I think (_think_) I've come up with a solution.

Comment (jafeltra) [reply src/export/StructureDefinitionExporter.ts]: After reviewing these changes for a full day, they look good to m!

**Review by jafeltra — APPROVED** [review]

**Review by julianxcarter — APPROVED** [review]

**Pull request merged**